apps/docs/src/guide/encoding/encode-and-decode.md

```diff
@@ -0,0 +1,21 @@
+# Encode and Decode
+
+Encoding is the process of transforming raw data into bytecode as per a set of rules. Decoding is the reverse process, transforming bytecode back to raw data. The SDK provides the `AbiCoder` class to encode and decode data.
+
+It has 3 static methods: `encode`, `decode`, and `getCoder`, `encode` and `decode` describe the aforementioned processes, while `getCoder` returns an instance of the internal coder required to serialize the passed type, and is used internally by the `encode` and `decode` methods. All methods expect you to pass the [ABI](https://docs.fuel.network/docs/specs/abi/json-abi-format/) and ABI Argument as function parameters, in order to deduce the specific type coders that will be required to parse the data.
+
+Imagine we are working with the following script that returns the sum of two `u32` integers:
+
+<<< @/../../docs-snippets/test/fixtures/forc-projects/sum-script/src/main.sw#encode-and-decode-1{rust:line-numbers}
+
+It will produce the following ABI:
+
+<<< @/../../docs-snippets/src/guide/encoding/encode-and-decode.test.ts#encode-and-decode-2{ts:line-numbers}
```

Comment:
Currently, we don't support using a JSON file as a code snippet, correct?

Reply:
Correct, it is invalid JSON.

Reply:
TIL ^^ `forc build` does not generate `.jsonc` though, and my main reason for doing it in a snippet is to keep the AVI in sync 🤔

Reply:
https://github.com/FuelLabs/fuels-ts/pull/2405/commits/948c714e2aca05653ce58e573b5b972d9d2e35e3

Reply:
And https://github.com/FuelLabs/fuels-ts/pull/2405/commits/3d958b4d61985b9d2b4f95983a32eff304342d18